ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/router_bgp.py

```diff
@@ -292,56 +324,68 @@ def _create_neighbor(self, name: str, peer_group: str, remote_as: str = None) ->
 
         return neighbor
 
-    def _neighbors(self) -> dict | None:
+    def _neighbors(self) -> list | None:
         """ """
-        neighbors = {}
+        neighbors = []
 
         if self._overlay_routing_protocol == "ebgp":
             for route_server, data in natural_sort(self._evpn_route_servers.items()):
-                neighbor = self._create_neighbor(route_server, self._peer_group_evpn_overlay_peers, remote_as=data["bgp_as"])
+                neighbor = {
+                    "ip_address": data["ip_address"],
+                    **self._create_neighbor(route_server, self._peer_group_evpn_overlay_peers, remote_as=data["bgp_as"]),
```

Comment:
if you add an `ip_address` argument to `_create_neighbor` I think most of these could be simplified.